src/Parallel/centrality/pagerank.jl

```diff
@@ -8,30 +8,35 @@ function pagerank(
     # indegree(g, v) is estimated run-time to iterate over inneighbors(g, v)
     partitions = LightGraphs.optimal_contiguous_partition(indegree(g), nthreads(), nv(g))
 
-    # collect dangling nodes
-    dangling_nodes = [v for v in vertices(g) if outdegree(g, v) == 0]
+    α_div_outdegree = Vector{Float64}(undef,nv(g))
+    dangling_nodes = Vector{U}()
+    for v in vertices(g)
+        if outdegree(g, v) == 0
+            push!(dangling_nodes, v)
+        end
+        α_div_outdegree[v] = (α/outdegree(g, v))
+    end
+
     N = Int(nv(g))
     # solution vector and temporary vector
     x = fill(1.0 / N, N)
     xlast = copy(x)
-    # personalization vector
-    p = fill(1.0 / N, N)
-    # adjustment for leaf nodes in digraph
-    dangling_weights = p
     for _ in 1:n
```

Comment:
There are a lot of array accesses here, so maybe add an `@inbounds` in front of the loop, but verify that there all the array accesses are inbounds first.